src/TraceEvent/TraceLog.cs

```diff
@@ -8622,6 +8636,10 @@ void IFastSerializable.FromStream(Deserializer deserializer)
                     codeAddressInfo.moduleFileIndex = (ModuleFileIndex)deserializer.ReadInt();
                     codeAddressInfo.methodOrProcessOrIlMapIndex = deserializer.ReadInt();
                     deserializer.Read(out codeAddressInfo.InclusiveCount);
+
+                    // Version 1
+                    codeAddressInfo.optimizationTier = (OptimizationTier)deserializer.ReadByte();
```

Comment:
The one thing that I'm slightly concerned about is that this is a breaking change for older etlx files.  This won't hurt PerfView as it separates etlx files by version, but it could hurt TraceEvent users if they use TraceLog and keep the etlx files around or transport them as a format.

It's not clear to me that there is an easy way to address this either though, as it's a per-method field, so you won't be able to use Deserializer.WriteTagged and TryReadTagged.

Let's leave this as is, but just be aware that if this becomes an issue we may need to address it.

Reply:
I was thinking about using a tagged value instead, I suppose the etlx would either always have or always not have optimization tiers for code addresses, so maybe it could work, but I was inclined to keep the size overhead as small as possible since there would be a lot of code addresses, though I didn't see much noticeable effect on size in traces over relatively larger apps compared to baseline whether using the current method or tagging. I could switch it to tagging if you think it might save some grief.

Reply:
I'm not sure you actually can tag it because tagging really only adds value at a per-type level, where the type implements IFastSerializable.  The class in question here is basically a list of them, so I don't think you can do this without restructuring things, and thus having a breaking change anyway.  I wouldn't worry about it at this point.

Reply:
> Another option which would give the best of both of the above options is to read/write a tagged value before code address infos to indicate a code address info version, and use that to look for or skip optimization tiers for each code address info. I'll give it a try.

> That might work since I believe that if it fails to read the tagged value it will back the stream up. Let me know if that causes a bunch of grief and we can re-evaluate.

I remember now that there were also other issues with serialization compatibility that were causing old etlx files to be incompatible anyway and I hadn't found an easy way to solve those, so I hadn't bothered to maintain compatibility here. I don't remember the details now, will do a scan.

Reply:
Ok, sounds good.

Reply:
Looks like this is the only issue currently. Fixed with the latest commit, it can now load etlx files created before this change.